### PR TITLE
Add STM32 generator

### PR DESCRIPTION
**Pull request by dbrgn**
Based on data from https://github.com/LibrePCB/stm32-pinout, based on initial work by @ouabache.

### CODE REVIEW

**Comment by dbrgn** [comment]
Current blocker: LibrePCB doesn't like parentheses in signal names.

    [EXCEPTION] Invalid identifier: 'PA13 (JTMS/SWDIO)' (../../../../libs/librepcb/common/circuitidentifier.h:54)
    [EXCEPTION] File parse error: Invalid identifier: 'PA13 (JTMS/SWDIO)'

**Comment by ouabache** [comment]
I parsed out everything after the space. That seemed to work

**Comment by dbrgn** [comment]
Also TODO: Group components by GPIO version, not purely by pinout.

This results in more variants, but is more future-proof if we add support for AFs.

**Comment by dbrgn** [comment]
> I parsed out everything after the space. That seemed to work

Stripping the text after pinout resulted in duplicates. I then noticed that some pins have remapped alternatives, these need to be ignored.

PR should be ready now!

File: generate_stm_mcu.py [hunk 1]
```diff
@@ -0,0 +1,822 @@
...
+        Created('2020-01-30T20:55:23Z'),
+        Deprecated(False),
+        cmpcat,
+        SchematicOnly(False),
+        DefaultValue('{{PARTNUMBER or DEVICE or COMPONENT}}'),
+        Prefix('U'),
+    )
+
+    # Add signals
+    signals = sorted({pin.name for pin in mcu.pins})
```
[the remark this answers is not in the document]
Comment (dbrgn) [reply generate_stm_mcu.py]: Sure! I updated both the generator and the library PR.

**Comment by dbrgn** [comment]
@ubruhin Thanks for your two assertion commits, I added them to this branch!

**Review by ubruhin — APPROVED** [review]

**Pull request merged**